pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -280,6 +281,10 @@ where
         FAILED_DOWNLOAD_WARN_THRESHOLD,
         FAILED_REMOTE_OP_RETRIES,
         description,
+        // TODO: pass a cancellation token into this function
+        backoff::Cancel::new(CancellationToken::new(), || -> DownloadError {
```

Comment:
NB: I think this function _is_ always running inside task manager, no?
Going through direct callers:
- download_index_part: yup, runs inside tenant load / attach task; the `upload_scheduling` is the only transitive caller that doesn't use task_mgr. maybe cfg(test) this here?
- list_remote_timelines: runs inside attach task 
- download_layer_file -> download_layer_file -> download_remote_layer runs in task_mgr task

So, I think using task_mgr shutdown_token() here _would_ work.
Of course it's preferable to plumb through the shutdown token explicitly through the call graph.
Are you planning to do that in a future PR?
If not, please create a follow-up issue from this comment.

Reply:
That's what I thought too, but it is not always used inside a task_mgr task:
```
   7: core::result::unwrap_failed
             at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/result.rs:1651:5
   8: pageserver::task_mgr::shutdown_token
             at /home/neon/neon/pageserver/src/task_mgr.rs:573:5
   9: pageserver::tenant::remote_timeline_client::download::download_retry::{{closure}}
             at /home/neon/neon/pageserver/src/tenant/remote_timeline_client/download.rs:285:30
  10: pageserver::tenant::remote_timeline_client::download::download_index_part::{{closure}}
             at /home/neon/neon/pageserver/src/tenant/remote_timeline_client/download.rs:256:6
  11: <pageserver::metrics::MeasuredRemoteOp<F> as core::future::future::Future>::poll
             at /home/neon/neon/pageserver/src/metrics.rs:1414:27
  12: pageserver::tenant::remote_timeline_client::RemoteTimelineClient::download_index_file::{{closure}}
             at /home/neon/neon/pageserver/src/tenant/remote_timeline_client.rs:460:10
  13: pageserver::tenant::Tenant::attach::{{closure}}::{{closure}}
             at /home/neon/neon/pageserver/src/tenant.rs:661:26
```

The issue for following up is https://github.com/neondatabase/neon/issues/5066 -- I suspect this will make sense to do at the same time as reorganizing mgr.rs